doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
++---------------------+------------------------------------------------------------------+----------------------------------------------------------------+
+| nrf700x_bh_wq       | :kconfig:option:`CONFIG_NRF700X_BH_WQ_STACK_SIZE`                | nRF700x Wi-Fi driver work queue                                |
++---------------------+------------------------------------------------------------------+----------------------------------------------------------------+
+| nrf700x_intr_wq     | :kconfig:option:`CONFIG_NRF700X_IRQ_WQ_STACK_SIZE`               | Interrupts processing generated by the nRF700X Wi-Fi radio     |
++---------------------+------------------------------------------------------------------+----------------------------------------------------------------+
+| mbox_wq             | :kconfig:option:`CONFIG_IPC_SERVICE_BACKEND_RPMSG_WQ_STACK_SIZE` | Inter Processor Communication                                  |
+|                     |                                                                  | (For multi-processors targets only)                            |
++---------------------+------------------------------------------------------------------+----------------------------------------------------------------+
+
+Heap size
+---------
+
+The Matter application defines the static size of the heap used by all memory allocations.
+This configuration ignores the :kconfig:option:`CONFIG_HEAP_MEM_POOL_SIZE` Kconfig option value in the application configuration.
+The static size is determined by the :kconfig:option:`CONFIG_CHIP_MALLOC_SYS_HEAP` and :kconfig:option:`CONFIG_CHIP_MALLOC_SYS_HEAP_OVERRIDE` Kconfig options.
+You can set them to ``n`` to start using a dynamic heap size on your Matter device.
+
+The default static heap size means that you can define the maximum heap size for your application by setting the :kconfig:option:`CONFIG_CHIP_MALLOC_SYS_HEAP_SIZE` Kconfig value.
+You can also adjust the heap dedicated for mbedTLS purposes by setting the :kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE` Kconfig option value.
+
+Memory profiling troubleshooting
+================================
+
+While manipulating memory sizes, you may experience various bugs and faults on your device.
+Many of these issues can be caused by incorrect stack or heap configurations.
+In this section, you will find some useful tips that you can use when setting new values for memory sizes, especially if you regularly notice problems with your application.
+
+- Check the thread name from which the fault originates:
+
+  If you notice a fault on your Matter device, analyze the logs to find the thread that was executing when the error occurred.
+  Sometimes, if there is not enough space in a specific thread, a crash may occur.
+  In such cases, you may consider increasing the size of the stack dedicated to that thread.
+  You can search through the table above and change the value of the specific Kconfig option.
```

Comment:
```suggestion
  Find the thread in the table above and change the value of the corresponding Kconfig option.
```
Could also link to the table directly. You'll need to add a ref target like is used for headings, and then use "the :ref:`thread and Kconfig option <ref target>` table".